src/dosext/mfs/mfs.c

```diff
@@ -2554,15 +2554,38 @@ RedirectDisk(int dsk, char *resourceName, int ro_flag)
   cds_base = MK_FP32(cdsfarptr.segment, cdsfarptr.offset);
 
   /* see if drive is in range of valid drives */
-  if(dsk < 0 || dsk > lol_last_drive(lol)) return 1;
+  if (dsk < 0 || dsk > lol_last_drive(lol))
+    return 1;
 
   cds = drive_cds(dsk);
 
   /* see if drive is already redirected */
-  if(cds_flags(cds) & CDS_FLAG_REMOTE) return 2;
+  if (cds_current_path(cds)[0] && cds_flags(cds) & CDS_FLAG_REMOTE) {
+    Debug0((dbg_fd, "RedirectDisk drive already redirected\n"));
+    Debug0((dbg_fd, "  cur_path is '%s'\n", cds_current_path(cds)));
+    cds_flags_str = cds_flags_to_str(cds_flags(cds));
+    if (cds_flags_str) {
```

Comment:
Do you really need this?
Just for debugging?
Maybe dumping the flags is enough? :)
Otherwise please make it into a debug
macro, it is not good to clutter the code
with debug code like this.